plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/mypy/MypyAnalysis.java

```diff
@@ -207,29 +243,46 @@ public void afterRunProcess(String output, String errors, IExternalCodeAnalysisS
                         markerSeverity = IMarker.SEVERITY_INFO;
                     }
                     if (markerSeverity != -1) {
-                        line = Integer.parseInt(outputLine.substring(m.start(2), m.end(2)));
+                        line = Integer.parseInt(outputLine.substring(m.start(2), m.end(2))) - 1;
                         column = Integer.parseInt(outputLine.substring(m.start(3), m.end(3))) - 1;
                         messageId = "mypy";
                         message = outputLine.substring(m.start(5), m.end(5)).trim();
+
+                        IRegion region = null;
+                        try {
+                            region = document.getLineInformation(line);
+                        } catch (Exception e) {
+                        }
+                        if (region != null && document != null) {
+                            boolean foundKey = false;
+                            for (LineCol lineCol : lineColToMessage.keySet()) {
```

Comment:
This doesn't seem a good way to get it. Rather, since you're working in a map, you should do 2 things:

1. Make sure that your key has `equals()/hashCode()` based on the important keys (use the `Shift+Alt+S` in the class and select the related option).

2. To get the value from the map you should do something as:

```
MessageInfo messageInfo = lineColToMessage.get(new LineCol(line, col));
if(messageInfo == null){
    // create
}else{
    // just append msg
}
```